pkg/testutil/framework.go

```diff
@@ -307,6 +315,182 @@ func (f *Framework) SetupServiceCluster(ctx context.Context, cl *RecordingClient
 	return serviceCluster
 }
 
+func (f *Framework) KubeCarrierOperatorCheck(
+	ctx context.Context,
+	t *testing.T,
+	managementClient *RecordingClient,
+	managementScheme *runtime.Scheme,
+) {
+	namePrefix := constants.KubeCarrierDefaultName + "-operator"
+	namespaceName := constants.KubeCarrierDefaultNamespace
+
+	f.componentCheck(ctx, managementClient, managementScheme, t, namePrefix, []runtime.Object{
+		&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-manager", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		// Webhook Service
+		&corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-webhook-service", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		// ClusterRoleBinding
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-manager-rolebinding", namePrefix),
+			},
+		},
+		// ClusterRole
+		&rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-manager-role", namePrefix),
+			},
+		},
+		// RoleBinding
+		&rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-leader-election-rolebinding", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		// Role
+		&rbacv1.Role{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-leader-election-role", namePrefix),
+				Namespace: namespaceName,
+			},
+		},
+		&apiextensionsv1.CustomResourceDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "accounts.catalog.kubecarrier.io",
+			},
+		},
+	})
+}
+
+func (f *Framework) E2EOperatorCheck(
```

Comment:
The same objection as in KubecarrierCheck. Also as far as I can see this is only called in a single place --> installation/e2e-operator; thus it would be better inlining it there instead of bloating this file even more. The installation/e2e-operator has 10lines...this file is become humongous.